### DISCUSSION TITLE
Installation File for SoftFever

### DISCUSSION THREAD

**Discussion by erebus04** [OrcaSlicer/OrcaSlicer#77] [General]
There are some users of 3D slicers who have problems when only one ZIP file is provided for the software.

My question would now be whether you see a need to provide an installation in the form of a Setup.exe here?

I would create and provide this for the respective release.

**Comment by SoftFever** [comment]
This would be great!
Can the installation package creation script be called from `build_release.bat`?
I plan to set up a CICD pipeline to streamline the build process.

**Comment by erebus04** [comment]
There is a possibility to integrate it into the CI/CD pipeline.
I would create it with InnoSetup.
In addition, the nuget package "Tools.InnoSetup" would have to be added to the Visual Studio project so that the build works via the command line.

There is an older guide though with Azure as the pipeline.

https://github.com/khelifaadm/InnoSetupSample

In the first step, I would build the release by hand to be able to test it extensively.
After that, it could be integrated into the pipeline.

**Reply by SoftFever** [reply]
Great.
Is it possible to install fonts as well?
https://github.com/SoftFever/BambuStudio-SoftFever/tree/SoftFever/resources/fonts

**Reply by erebus04** [reply]
Yes that's possible.

**Reply by erebus04** [reply]
I have applied for a code signing certificate so that the installer is not blocked by Windows.
As soon as I have the certificate I will provide you with a test build.

**Comment by SoftFever** [comment]
You are awesome!Sent from my iPhoneOn 1 Jan 2023, at 9:30 PM, erebus04 ***@***.***> wrote:﻿
I have applied for a code signing certificate so that the installer is not blocked by Windows.
As soon as I have the certificate I will provide you with a test build.

**Comment by erebus04** [comment]
I have a first alpha version of the installer.
Unfortunately it is too big to upload directly to Github.

There are two things I had to change currently.
1. the whole thing does not work with Inno Setup as intended. I have therefore been looking for an alternative. I then wrote to the manufacturer of AKInstaller. He would provide his software for free, if he is mentioned and linked in the project.
Would that be OK for you? @SoftFever 

2. the certificate for the signing is issued on my name. Verification had to be done by passport etc. It is a cloud hosted certificate that requires a Generated Token from my APP for each access. 

Due to these two circumstances, automatic generation will not work at this time. I would take care of the generation. 
If that would be ok for you?

Is it possible to upload larger files in the releases?

Here is the link to the installer:

https://drive.proton.me/urls/DZJY0001S0#aWSS6kUxX5KH

**Comment by SoftFever** [comment]
@erebus04 
Thank you so much for exploring this!

Unfortunately, the manual process makes it impractical. 
I'm afraid it will make the releasing process a nightmare, as it involves multiple people from different time zones.

Again, I really appreciate everything you have done!

**Reply by erebus04** [reply]
@SoftFever 
did you mention me on Discord?
Can I reach you through it?
I have invested some time and found a way to make the build via the ci/cd.
For this we should match some topics that are not public.

**Comment by SoftFever** [comment]
No, I didn’t.A bit busy this week:( will ping you when I got time, probably this weekend.Sent from my iPhoneOn 9 Jan 2023, at 9:52 PM, Michael Birkler ***@***.***> wrote:﻿
@SoftFever
did you mention me on Discord?
Can I reach you through it?
I have invested some time and found a way to make the build via the ci/cd.
For this we should match some topics that are not public.
